preferences-commons/src/main/java/no/sikt/nva/person/preferences/commons/model/PersonPreferencesDao.java

```diff
@@ -1,69 +1,107 @@
-                   .withPromotedPublications(this.promotedPublications)
-                   .withCreatedDate(this.created)
-                   .withModifiedDate(this.modified);
+    @DynamoDbIgnore
+    @Override
+    public PersonPreferencesDao upsert(DataAccessService<PersonPreferencesDao> service) throws NotFoundException {
+        service.persist(this);
+        return fetch(service);
+    }
+
+    @DynamoDbIgnore
+    @Override
+    public PersonPreferencesDao fetch(DataAccessService<PersonPreferencesDao> service) throws NotFoundException {
+        return service.fetch(this);
     }
 
     public static class Builder {
+        public static final String PROMOTED_PUBLICATIONS = "PromotedPublications";
+        private URI id;
+        private String type;
+        private List<URI> promotedIds;
+        private Instant createdInstant;
+        private Instant modifiedInstant;
 
-        private URI personId;
-        private List<URI> promotedPublications;
-        private Instant created;
-        private Instant modified;
+        private Builder() {
+        }
 
-        public Builder withPersonId(URI personId) {
-            this.personId = personId;
+        public Builder withId(URI withId) {
+            this.id = withId;
             return this;
         }
 
-        public Builder withPromotedPublications(List<URI> promotedPublications) {
-            this.promotedPublications = promotedPublications;
+        public Builder withType(String withType) {
+            this.type = withType;
             return this;
         }
 
-        public Builder withCreatedDate(Instant created) {
-            this.created = created;
+        public Builder promotedPublications(List<URI> promotedPublications) {
+            this.promotedIds = promotedPublications;
             return this;
         }
 
-        public Builder withModifiedDate(Instant modified) {
-            this.modified = modified;
+        public Builder created(Instant created) {
+            this.createdInstant = created;
             return this;
         }
 
-        public PersonPreferencesDao fromDynamoFormat(Map<String, AttributeValue> map) {
-            return attempt(() -> JsonUtils.dynamoObjectMapper.readValue(ItemUtils.toItem(map).toJSON(),
-                                                                        PersonPreferencesDao.class))
-                       .orElseThrow();
+        public Builder modified(Instant modified) {
+            this.modifiedInstant = modified;
+            return this;
         }
 
         public PersonPreferencesDao build() {
-            return new PersonPreferencesDao(personId, promotedPublications, created, modified);
+            if (modifiedInstant == null) {
+                modified(Instant.now());
+            }
+
+            if (createdInstant == null) {
+                created(modifiedInstant);
+            }
+
+            if (type == null) {
+                withType(PROMOTED_PUBLICATIONS);
+            }
```

Comment:
This should be moved into an extendtion of `DynamoDbEnhancedClient`. (this is basical db internals)